### DISCUSSION TITLE
Euler angle in MTEX

### DISCUSSION THREAD

**Discussion by jiantangethz** [mtex-toolbox/mtex#1941] [Ask Anything] [answered]
Hi MTEX community,

I notice that the Euler angles in the ebsd.orientations are different from my input file.  For example, MTEX shows an Euler angle of (279.7,128.5,219.5) for the example grain in the following figure, however, it should be (260.5, 51.29, 39.32) in my input ctf file.  Could you help me to understand why does this happen?  Besides, what is the default range of three Euler angles in MTEX (e.g. can i import an negative Euler angle)?
![image](https://github.com/mtex-toolbox/mtex/assets/97729213/374cd075-d350-47df-a5ac-f5eae058f94d)

My ctf file is output by OIM EBSD software. They use Bunge Euler angle (ZXZ) and the ranges of three Euler angles are (0,360), (0,90), and (0,90).  I transform it into a txt file here.
[Exp.txt](https://github.com/mtex-toolbox/mtex/files/12520353/Exp.txt)

And the following are the code I use in MTEX:
    % crystal symmetry
    CS = {... 
      'notIndexed',...
      crystalSymmetry('m-3m', [3.6 3.6 3.6], 'mineral', 'Nickel', 'color', [0.53 0.81 0.98])};
    
    % plotting convention
    setMTEXpref('xAxisDirection','west');
    setMTEXpref('zAxisDirection','outOfPlane');

    fname='Exp.ctf';
    % create an EBSD variable containing the data
    ebsd = EBSD.load(fname,CS,'interface', extractAfter(fname,'.'),...
      'convertEuler2SpatialReferenceFrame');
    
    
    [grains,ebsd.grainId] = calcGrains(ebsd('Nickel'),'angle',10*degree);
    grains = smooth(grains,3);
    figure
    ipfKey = ipfColorKey(ebsd('Nickel'));
    ipfKey.inversePoleFigureDirection = vector3d.Y;   % colors follow orientations in y direction
    colors = ipfKey.orientation2color(ebsd('Nickel').orientations);
    plot(ebsd('Nickel'),colors)

Thank you very much :)

**Comment by ralfHielscher** [comment, marked as answer]
This is due to your option `'convertEuler2SpatialReferenceFrame'` which changes the Euler angles.
Read more about this at https://mtex-toolbox.github.io/EBSDReferenceFrame.html

Ralf.

**Reply by jiantangethz** [reply]
Dear Ralf,

Thanks for your answer.  Have a nice day!

All the best
Jian
